agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -115,9 +142,35 @@ public static PolicyRestrictor lookupPolicyRestrictor(String pLocation) throws I
                 is = RestrictorFactory.class.getResourceAsStream(path);
             }
         } else {
-            URL url = new URL(pLocation);
-            is = url.openStream();
+            URLConnectionResult urlConnectionResult = buildUrlConnection(pLocation);
```

Comment:
I wonder why you need this temporary result object anyway. Couldn't you simply return the UrlConnection itself ?

If it is only for the test, I would prefer another strategy. Put the parsing stuff in an utility class and test this since this is the only critical part here. Change the original code for better testing is a smell per se.

I mean you could have a static 'String[] extractUrlAndAuthentication(url)' method which returns both parts url, and encoded base64 string in a two element array. This could be then tested. Ok, not perfect, too, but you really dont need this result object imo.